src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let stateCopy = Object.assign({}, state);
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        stateCopy = Object.assign({}, stateCopy, key.extraData);
+        break;
+
+      case 'removeProperties':
+        stateCopy = Object.assign({}, stateCopy);
+
+        for (const rem of key.keysToRemove) {
+          delete stateCopy[rem];
+        }
+        break;
+        
+      case 'clear':
+        stateCopy = {};
+        break;
+    }
```

Comment:
[CODE STYLE]: A `default` case should be added to the `switch` statement for error handling. It helps to catch any unexpected `type` values and can be used to throw an error or handle the case appropriately.